services-core/src/gas_price.rs

```diff
@@ -62,45 +38,36 @@ pub async fn create_priority_estimator(
                 if !is_mainnet(&network_id) {
                     return Err(anyhow!("EthGasStation only supports mainnet"));
                 }
-                estimators.push(Box::new(ethgasstation::EthGasStation::new(
-                    http_factory.create()?,
-                )))
+                estimators.push(Box::new(EthGasStation::new(http_factory.create()?)))
             }
             GasEstimatorType::GasNow => {
                 if !is_mainnet(&network_id) {
                     return Err(anyhow!("GasNow only supports mainnet"));
                 }
-                estimators.push(Box::new(gasnow::GasNow::new(http_factory.create()?)))
+                estimators.push(Box::new(GasNow::new(http_factory.create()?)))
             }
             GasEstimatorType::GnosisSafe => estimators.push(Box::new(
-                gnosis_safe::GnosisSafeGasStation::with_network_id(
-                    &network_id,
-                    http_factory.create()?,
-                )?,
+                GnosisSafeGasStation::with_network_id(&network_id, http_factory.create()?)?,
             )),
             GasEstimatorType::Web3 => estimators.push(Box::new(web3.clone())),
         }
     }
-    Ok(Arc::new(priority::PriorityGasPrice::new(estimators)))
+    Ok(Arc::new(PriorityGasPrice::new(estimators)))
 }
 
 fn is_mainnet(network_id: &str) -> bool {
     network_id == "1"
 }
 
 #[cfg(test)]
-mod tests {
-    use super::*;
-    use isahc::ResponseExt;
-
-    #[derive(Default)]
-    pub struct TestTransport {}
+use std::time::Duration;
+#[cfg(test)]
+mockall::mock! {
```

Comment:
Not sure if that is the best way to recreate a mockable version of the GasPriceEstimating trait in `core-services`. Alternatively we could export it from the `gas-estimation` crate gated behind a feature that `services-core` dev-depends on.

Reply:
I find it reasonable to not make any mock related choice in the new crate and instead let services_core mock it like you have implemented.